include/range/v3/utility/variant.hpp

```diff
@@ -67,21 +67,21 @@ namespace ranges
             template<>
             union variant_data<>
             {
-                void move(std::size_t, variant_data &&) const
+                [[noreturn]] void move(std::size_t, variant_data &&) const
                 {
                     RANGES_ASSERT(false);
                 }
-                void copy(std::size_t, variant_data const &) const
+                [[noreturn]] void copy(std::size_t, variant_data const &) const
                 {
                     RANGES_ASSERT(false);
                 }
-                bool equal(std::size_t, variant_data const &) const
+                [[noreturn]] bool equal(std::size_t, variant_data const &) const
                 {
                     RANGES_ASSERT(false);
                     return true;
                 }
```

Comment:
I don't know. I never liked this. One should never call this functions, right now this is only checked in debug mode. I would rather have something like a call to std::terminate(), but trackable. So that means either just 

    throw "you should never call this because.... at line....";

or maybe just always call:

   __builtin_debug_trap();

which triggers an assertion and will make your debugger stop at the right place.